src/utils.jl

```diff
@@ -134,14 +138,83 @@ julia> chunk(1:10, 3)
  5:8
  9:10
 
-julia> chunk(collect(1:10), 3)
-3-element Vector{SubArray{Int64, 1, Vector{Int64}, Tuple{UnitRange{Int64}}, true}}:
- [1, 2, 3, 4]
- [5, 6, 7, 8]
- [9, 10]
+julia> x = reshape(collect(1:20), (5, 4))
+5×4 Matrix{Int64}:
+ 1   6  11  16
+ 2   7  12  17
+ 3   8  13  18
+ 4   9  14  19
+ 5  10  15  20
+
+julia> xs = chunk(x, 2, dims=1)
+2-element Vector{SubArray{Int64, 2, Matrix{Int64}, Tuple{UnitRange{Int64}, Base.Slice{Base.OneTo{Int64}}}, false}}:
+ [1 6 11 16; 2 7 12 17; 3 8 13 18]
+ [4 9 14 19; 5 10 15 20]
+
+julia> xs[1]
+3×4 view(::Matrix{Int64}, 1:3, :) with eltype Int64:
+ 1  6  11  16
+ 2  7  12  17
+ 3  8  13  18
 ```
 """
-chunk(xs, n) = collect(Iterators.partition(xs, ceil(Int, length(xs)/n)))
+chunk(x, n::Int) = collect(Iterators.partition(x, ceil(Int, length(x) / n)))
+
+function chunk(x::AbstractArray, n::Int; dims::Int=ndims(x))
+    idxs = _partition_idxs(x, n, dims) 
+    [selectdim(x, dims, i) for i in idxs]
+end
+
+# Zygote errors if not iterating over collected partitions in [selectdim(x, dims, i) for i in ids] 
+function _partition_idxs(x, n, dims)
+    bs = ceil(Int, size(x, dims) / n)
+    collect(Iterators.partition(axes(x, dims), bs))
+end
+
+@non_differentiable _partition_idxs(x...)
```

Comment:
Is this non-diff _partition_idxs still needed once there's an explicit rule? If Zygote looks inside that it will see mutation and give up. 
```suggestion
function chunk(x::AbstractArray, n::Int; dims::Int=ndims(x))
    bs = ceil(Int, size(x, dims) / n)
    idxs = Iterators.partition(axes(x, dims), bs)
    [selectdim(x, dims, i) for i in idxs]
end
```

Reply:
`_partition_idxs` is used in nabla chunk as well. I agree making it `@non_differentiable` is superflous.